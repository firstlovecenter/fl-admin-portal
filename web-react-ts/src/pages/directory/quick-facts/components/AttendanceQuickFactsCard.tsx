import React from 'react'
import { Badge } from 'react-bootstrap'
import '../QuickFacts.css'
import { getPercentageChange } from './quick-fact-utils'

export interface AttendanceQuickFactsProps {
  attendanceDetails: {
    churchType: string
    cardType: string
    leadersName: string
    churchName: string
    churchAvgAttendanceThisMonth: number | string
    avgHigherLevelAttendanceThisMonth: number | string
    higherLevelName: string
  }[]
}

const AttendanceQuickFactsCard = (props: AttendanceQuickFactsProps) => {
  const details = props?.attendanceDetails[0]

  const percentageRiseOrFall = getPercentageChange(
    details?.churchAvgAttendanceThisMonth as number,
    details?.avgHigherLevelAttendanceThisMonth as number
  )

  const getBadgeBackground = () => {
    if (percentageRiseOrFall >= 0) return 'green'
    return 'red'
  }

  const getBadgeColor = () => {
    if (percentageRiseOrFall >= 0) return 'badge-percentage-green'
    return 'badge-percentage-red'
  }

  return (
    <div className="w-100 text-center quick-fact-card">
      <div className="church-text">{details?.churchType}</div>
      <div className="stat-text ">
        Average Weekday <br />
        {details?.cardType}
      </div>
      <div className="leader-text">{details?.leadersName}</div>
      <div className="branch-text">
        {details?.churchName + ' ' + details?.churchType}
      </div>
      <div className="facts-number">
        {details?.churchAvgAttendanceThisMonth === 'null'
          ? '--'
          : details?.churchAvgAttendanceThisMonth}
      </div>
      <div>
        <Badge
          bg={`${getBadgeBackground()}`}
          className={`${getBadgeColor()} mt-auto`}
        >
          {percentageRiseOrFall >= 0 ? '+' : ''}
          {percentageRiseOrFall}%
        </Badge>
      </div>
      <hr className="separator" />
      <div className="facts-number text-center">
        {details?.avgHigherLevelAttendanceThisMonth === 'null'
          ? '--'
          : details?.avgHigherLevelAttendanceThisMonth}
      </div>
      <div className="average-text">
        Average {details?.churchType} <br /> Attendance
      </div>
      <div className="higher-church-text">{details?.higherLevelName}</div>
    </div>
  )
}

export default AttendanceQuickFactsCard
