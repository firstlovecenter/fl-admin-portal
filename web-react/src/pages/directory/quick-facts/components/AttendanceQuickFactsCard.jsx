import React from 'react'
import { Badge } from 'react-bootstrap'
import '../QuickFacts.css'

const AttendanceQuickFactsCard = (props) => {
  const details = props?.details[0]

  const getPercentageChange = (avgAttendance, avgHigherLevelAttendance) => {
    var diff = avgAttendance - avgHigherLevelAttendance
    if (isNaN(diff)) return '--'
    return Math.round((diff / avgAttendance) * 100)
  }

  const percentageRiseOrFall = getPercentageChange(
    details?.churchAvgAttendanceThisMonth,
    details?.avgHigherLevelAttendanceThisMonth
  )

  const getBadgeBackground = () => {
    if (percentageRiseOrFall >= 0) return 'green'
    return 'red'
  }

  const getBadgeColor = () => {
    if (percentageRiseOrFall >= 0) return 'badge-percentage-green'
    return 'badge-percentage-red'
  }

  return (
    <div className="w-100 text-center quick-fact-card">
      <div className="church-text">{details?.churchType}</div>
      <div className="stat-text ">
        Average Weekday <br />
        {details?.cardType}
      </div>
      <div className="leader-text">{details?.leadersName}</div>
      <div className="branch-text">
        {details?.churchName + ' ' + details?.churchType}
      </div>
      <div className="facts-number">
        {details?.churchAvgAttendanceThisMonth === 'null'
          ? '--'
          : details?.churchAvgAttendanceThisMonth}
      </div>
      <div>
        <Badge
          bg={`${getBadgeBackground()}`}
          className={`${getBadgeColor()} mt-auto`}
        >
          {percentageRiseOrFall >= 0 ? '+' : ''}
          {percentageRiseOrFall}%
        </Badge>
      </div>
      <hr className="separator" />
      <div className="facts-number text-center">
        {details?.avgHigherLevelAttendanceThisMonth === 'null'
          ? '--'
          : details?.avgHigherLevelAttendanceThisMonth}
      </div>
      <div className="average-text">
        Average {details?.churchType} <br /> Attendance
      </div>
      <div className="higher-church-text">{details?.higherLevelName}</div>
    </div>
  )
}

export default AttendanceQuickFactsCard
