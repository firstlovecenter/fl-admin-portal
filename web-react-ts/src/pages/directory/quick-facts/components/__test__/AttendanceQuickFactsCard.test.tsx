import { render, cleanup } from '@testing-library/react'
import React from 'react'
import AttendanceQuickFactsCard from '../AttendanceQuickFactsCard'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

const attendanceDetails = [
  {
    churchType: 'Bacenta',
    cardType: 'Attendance',
    leadersName: 'Edem Ahadzi',
    churchName: 'Madina Bacenta',
    churchAvgAttendanceThisMonth: '23',
    avgHigherLevelAttendanceThisMonth: '63',
    higherLevelName: 'Madina Council',
  },
]

afterEach(cleanup)

it('renders without crashing', () => {
  render(<AttendanceQuickFactsCard attendanceDetails={attendanceDetails} />)
})

it('renders Attendance Card with correct details', () => {
  const { getByTestId } = render(
    <AttendanceQuickFactsCard attendanceDetails={attendanceDetails} />
  )
  expect(getByTestId('attendanceCard')).toHaveTextContent('Attendance')
})

it('matches snapshot', () => {
  const tree = renderer
    .create(<AttendanceQuickFactsCard attendanceDetails={attendanceDetails} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
