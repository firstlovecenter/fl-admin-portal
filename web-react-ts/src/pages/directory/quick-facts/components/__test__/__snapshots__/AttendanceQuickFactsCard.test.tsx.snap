// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="w-100 text-center quick-fact-card"
  data-testid="attendanceCard"
>
  <div
    className="church-text"
  >
    Bacenta
  </div>
  <div
    className="stat-text "
  >
    Average Weekday 
    <br />
    Attendance
  </div>
  <div
    className="leader-text"
  >
    Edem Ahadzi
  </div>
  <div
    className="branch-text"
  >
    Madina Bacenta Bacenta
  </div>
  <div
    className="facts-number"
  >
    23
  </div>
  <div>
    <span
      className="badge-percentage-red mt-auto badge bg-red"
    >
      -63
      %
    </span>
  </div>
  <hr
    className="separator"
  />
  <div
    className="facts-number text-center"
  >
    63
  </div>
  <div
    className="average-text"
  >
    Average 
    Bacenta
     
    <br />
     Attendance
  </div>
  <div
    className="higher-church-text"
  >
    Madina Council
  </div>
</div>
`;
